### ISSUE TITLE
`jsdoc/match-name` do not match correctly when default value of `@template` contains spaces

### ISSUE THREAD

**Issue by Arkellys** [gajus/eslint-plugin-jsdoc#1233]
## Expected behavior

I expect the rule [`jsdoc/match-name`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/match-name.md) to ignore the default param passed to `@template` when testing the name, whether or not the value contains spaces. It works correctly when the same rules are applied on `@property`.

## Actual behavior

Currently [`jsdoc/match-name`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/match-name.md) fails when the default value passed to a `@template` contains space.

## ESLint Config

```js
"jsdoc/match-name": ["warn", {
   match: [
    {
      allowName: "/^[A-Z]{1}$/",
      message: "The name should be a single capital letter.",
      tags: ["template"]
    }
  ]
}]
```

Full config available [here](https://github.com/Arkellys/eslint-config-arklint/blob/main/extensions/jsdoc.js).

## ESLint sample

This fails:

```js
/**
 * @template {string} [T=typeof FOO]  ->  The name should be a single capital letter. eslint(jsdoc/match-name)
 * @typedef {object} Test
 * @property {T} test
 */
```

These passes:

```js
/**
 * @template {string} [T="foo"]
 * @typedef {object} Test
 * @property {T} test
 */
```

```js
/**
 * @template {string} T
 * @typedef {object} Test
 * @property {T} test
 */
```

Note that the space also messes up with the rule [`jsdoc/require-hyphen-before-param-description`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/require-hyphen-before-param-description.md#repos-sticky-header), and auto fix will do:

```js
@template [T=typeof - FOO]
```

## Environment

- Node version: 20.9.0
- ESLint version: 8.57.0
- `eslint-plugin-jsdoc` version: 48.2.6
